force-app/main/default/classes/RD2_ERecurringDonationsSelector.cls

```diff
@@ -50,34 +53,40 @@ public inherited sharing class RD2_ERecurringDonationsSelector {
+
+        Id currentContactId = this.currentContactId;
+        List<npe03__Recurring_Donation__c> recurringDonations = new List<npe03__Recurring_Donation__c>();
+        if(currentContactId != NULL) {
+            String queryString = 'SELECT npe03__Amount__c, '
+                                + 'npe03__Installment_Period__c, '
+                                + 'InstallmentFrequency__c, '
+                                + 'Day_of_Month__c, '
+                                + 'Status__c, '
+                                + 'CommitmentId__c, '
+                                + 'PaymentMethod__c, '
+                                + 'CardLast4__c, '
+                                + 'CardExpirationYear__c, '
+                                + 'CardExpirationMonth__c, '
+                                + 'ACH_Last_4__c, '
+                                + 'npe03__Last_Payment_Date__c, '
+                                + 'npe03__Installments__c, '
+                                + 'RecurringType__c, '
+                                + 'npe03__Date_Established__c, '
+                                + 'npe03__Next_Payment_Date__c '
+                                + 'FROM npe03__Recurring_Donation__c '
+                                + 'WHERE npe03__Contact__c = \'' + currentContactId + '\' ';
 
-		Id currentContactId = this.currentContactId;
-		List<npe03__Recurring_Donation__c> recurringDonations = new List<npe03__Recurring_Donation__c>();
-		if(currentContactId != NULL) {
-			recurringDonations = [  SELECT npe03__Amount__c,
-										npe03__Installment_Period__c,
-										InstallmentFrequency__c,
-										Day_of_Month__c,
-										Status__c,
-                                        CommitmentId__c,
-                                        PaymentMethod__c,
-                                        CardLast4__c,
-                                        CardExpirationYear__c,
-                                        CardExpirationMonth__c,
-                                        ACH_Last_4__c,
-                                        npe03__Last_Payment_Date__c,
-                                        npe03__Installments__c,
-										npe03__Date_Established__c,
-										RecurringType__c,
-										npe03__Next_Payment_Date__c
-									FROM npe03__Recurring_Donation__c
-									WHERE npe03__Contact__c = :currentContactId
-                  					AND RecurringType__c IN (:RD2_Constants.RECURRING_TYPE_OPEN, :RD2_Constants.RECURRING_TYPE_FIXED) 
-									WITH SECURITY_ENFORCED 
-									ORDER BY npe03__Next_Payment_Date__c DESC NULLS LAST];
-		}
+            if(elevateFilter == ELEVATE_RD) {
+                queryString += 'AND CommitmentId__c != NULL ';
+            } else if (elevateFilter == NONELEVATE_RD) {
+                queryString += 'AND CommitmentId__c = NULL ';
+            }
 
+            queryString += 'WITH SECURITY_ENFORCED ';
+            queryString += 'ORDER BY npe03__Next_Payment_Date__c DESC NULLS LAST';
```

Comment:
It looks like you're setting the queryString, but not actually calling the query?